File: gb-spring-backend/src/main/java/com/egroupbuy/model/Product.java
package com.egroupbuy.model;

import lombok.*;

import java.sql.Date;
import java.sql.Timestamp;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Product {
    int product_id;
    String title;
    Timestamp creation_date;
    String sell_style;
    String description;
    String image_urls;
    int brand_id;
    String material;
    Date delivery_date;
    String delivery_method;
    String delivery_region;
    int min_threshold;
    int stock_quantity;
    String stock_status;
    String status;
    double cost;
    int category_id;
    int seller_id;
    String seller_username;
    int likes;
    double regular_price;
    double sale_price;
    int display_position;
    String main_image;
    String brand_name;
    String category_name;
    Timestamp grpbuy_start;
    Timestamp grpbuy_end;
    int grpbuy_quantity;
    double grpbuy_regular_price;
    double grpbuy_sale_price;
    String grpbuy_status;
    int grpbuy_order_count;
    int package_grp_count;
    List<ProductItem> product_item_list;
    ReviewOverallScore review_overall_score;

}

File: gb-angular-frontend/src/app/interfaces/product.ts
import {Package} from './package';
import {ProductItem} from './product-item';
import {ReviewOverallScore} from './review-overall-score';

export interface Product {
  product_id: number;
  title: string;
  creation_date: string;
  sell_style: string;
  description: string;
  image_urls: string;
  brand_id: number;
  material: string;
  delivery_date: string;
  delivery_method: string;
  delivery_region: string;
  min_threshold: number;
  stock_quantity: number;
  stock_status: string;
  status: string;
  cost: number;
  regular_price?: number;
  sale_price?: number;
  category_id: number;
  seller_id: number;
  seller_username: string;
  likes: number;
  display_position: number;
  main_image: string;
  brand_name: string;
  category_name: string;
  grpbuy_start: string;
  grpbuy_end: string;
  grpbuy_quantity: number;
  grpbuy_regular_price: number;
  grpbuy_sale_price: number;
  grpbuy_status: string;
  grpbuy_order_count: number
  package_grp_count: number;
  product_item_list: ProductItem[];
  review_overall_score?: ReviewOverallScore;
}
